index.js

```diff
@@ -24,18 +24,20 @@ function MockDate(param) {
     if (param instanceof MockDate) {
       this.d = new _Date(param.d);
     } else if (typeof param === 'string') {
-      if (param.match(date_iso_8601_regex) || param.match(date_with_offset) || param.match(date_rfc_2822_regex) || param === '') {
+      if (param.match(date_iso_8601_regex) || param.match(date_with_offset) || param.match(date_rfc_2822_regex)) {
         this.d = new _Date(param);
-      } else if (param.match(local_date_regex)) {
+      } else if (param === '') {
+        this.d = new _Date(param);
+        this.invalid = true;
+      }
+      else if (param.match(local_date_regex)) {
         this.d = new _Date();
         this.fromLocal(new _Date(param.replace(' ', 'T') + 'Z'));
       } else {
         assert.ok(false, 'Unhandled date format passed to MockDate constructor: ' + param);
       }
-    } else if (typeof param === 'number') {
+    } else if (typeof param === 'number' || param == null) {
```

Comment:
Here is the null constructor support :)

Reply:
`param == null` will check if it's `undefined` too.

Reply:
Though it will, most linters will complain about that, and it's not clear from reading the code that that's what's going on, so I'd prefer you either add a comment or just be explicit and say `param === null || param === undefined`.

Reply:
Updated.